### PR TITLE
RATEST-67 Write Attachments page

### PR DESCRIPTION
**Pull request by achilep**
issue I work on : https://issues.openmrs.org/browse/RATEST-67

### CODE REVIEW

File: src/test/java/org/openmrs/contrib/qaframework/page/Attachmentspage.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    public Attachmentspage(Page parent) {
+        super(parent);
+    }
+
+    public void ClickOnAttachement() {
+        clickOn(ATTACHEMENT);
+    }
+
+    public void ClickOnClickOrDropAfile(){
```
Comment (sherrif10) [inline src/test/java/org/openmrs/contrib/qaframework/page/Attachmentspage.java:26]: @achilep  thanks this looks good to me, However , got some slight question. Assuming you make this class **ClickOnOrDropAfile** how does that look , then you can change the static file a relative name. its justa suggestion cc @k-joseph

Comment (sherrif10) [reply src/test/java/org/openmrs/contrib/qaframework/page/Attachmentspage.java:26]: Is there any possibility of adding screen shots for pages you have tested, i think this is quite tricky when it comes to testing such pages in ref-app or using standalone

**Comment by tendomart** [comment]
@achilep  did you see the previous comments from @sherrif10  ?

**Comment by achilep** [comment]
thanks, @tendomart  and @sherrif10. I will update the PR very soon.

File: src/test/java/org/openmrs/contrib/qaframework/page/Attachmentspage.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import java.util.List;
+import org.openmrs.uitestframework.page.Page;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+public class Attachmentspage extends Page {
+
+    public static final By ATTACHEMENT = By.id("attachments.attachments.visitActions.default");
+    public static final By CLICKORDROPAFILE = By.id("visit-documents-dropzone");
+    public static final By CAPTION = By.xpath("//*[@id=\"att-page-main\"]/div[1]/att-file-upload/div[2]/div/div[2]/textarea");
```
Comment (kaweesi) [inline src/test/java/org/openmrs/contrib/qaframework/page/Attachmentspage.java:12]: the problem with xpath through divs is that when interface changes come in, these get broken, figure out how to use attributes if you can please

**Comment by achilep** [comment]
please @k-joseph  
can you check XPath again? thanks

**Comment by sherrif10** [comment]
cc @achilep  Do you also try considering using Gherken Scenarios  Feel free to take alook at an [example here ](https://github.com/openmrs/openmrs-contrib-qaframework/pull/9/files)

**Comment by achilep** [comment]
Thanks @sherrif10

**Comment by achilep** [comment]
I suggest that one person write the feature file, the page, and the steps definition as a single PR instead of having them in several PR written by different person.
@k-joseph  @sherrif10

**Comment by sherrif10** [comment]
Sure We shall create a ticket for those features, is it fine to go ahead cc @k-joseph

**Comment by kaweesi** [comment]
@achilep  you still interested in completing this PR? please update it to address the reviews

**Comment by achilep** [comment]
yes, thanks @kaweesi, I will update the PR.

**Comment by kaweesi** [comment]
to avoid duplication of efforts, @achilep and @kdaud please have one of you work on the attachments feature. merge the feature file, page and steps into one change please and we can close the rest of the trailing PRs

**Comment by achilep** [comment]
nice suggestion @kaweesi, I will close these 2 PRs as they are all related to the attachment. https://github.com/openmrs/openmrs-contrib-qaframework/pull/8,  https://github.com/openmrs/openmrs-contrib-qaframework/pull/7. @kdaud can continue from  [here](https://github.com/openmrs/openmrs-contrib-qaframework/pull/11l)

**Comment by kdaud** [comment]
Alright Bebeto

**Pull request closed without merging**